### PR TITLE
ZCS-13071:  Drive module with shared files loss on sharee end if it is later moved to 10.0 and sharer has already been moved

### PR DESCRIPTION
**Pull request by shrutig0510**
**Problem:**
Drive module with shared files loss on sharee end if it is later moved to 10.0 and sharer has already been moved

**Fix:**
While exporting the sharee data, also export data of its remote shares from other users.
while importing sharee data, process these remote shares data on sharer end.

**Test-cases:**

1. sharer is moved first to 10.0 (all export and import done) and then sharee is moved to 10.0.
2. sharee is moved first to 10.0 (all export and import done) and then sharer is moved to 10.0.
3. sharer and sharee both moved to 10.0, but first sharer data is imported and then sharee data is imported.
4. sharer and sharee both moved to 10.0, but first sharee data is imported and then sharer data is imported.
In all the above cases, sharee remote files data is restored.

### CODE REVIEW

**Review by preshitaagrawal — APPROVED** [review]

File: store/src/java/com/zimbra/cs/mailbox/Mailbox.java [hunk 1]
```diff
@@ -9555,6 +9555,66 @@ public Document createDocument(OperationContext octxt, int folderId, ParsedDocum
         }
     }
 
+    public Document createDocument(OperationContext octxt, int folderId, ParsedDocument pd, MailItem.Type type,
```
Comment (vishnukantjangid) [inline store/src/java/com/zimbra/cs/mailbox/Mailbox.java:9502]: Use method overloading instead of creating the same duplicate method.

Comment (shrutig0510) [reply store/src/java/com/zimbra/cs/mailbox/Mailbox.java:9502]: This is method overloading only, introduced one new parameter at end as nodeId which will mostly be used for migration. Not touched already available methods so as to restrict the changes for migration mostly.
>public Document createDocument(OperationContext octxt, int folderId, String filename, String mimeType,
            String author, String description, InputStream data)

>public Document createDocument(OperationContext octxt, int folderId, ParsedDocument pd, MailItem.Type type,
            int flags, MailItem parent, CustomMetadata custom, boolean indexing, String nodeId)

Comment (vishnukantjangid) [reply store/src/java/com/zimbra/cs/mailbox/Mailbox.java:9502]: Yes, you can achieve the same thing without having duplicate methods. like 

```
public Document createDocument(OperationContext octxt, int folderId, ParsedDocument pd, MailItem.Type type,
             int flags, MailItem parent, CustomMetadata custom, boolean indexing) {
             createDocument(OperationContext octxt, int folderId, ParsedDocument pd, MailItem.Type type,
             int flags, MailItem parent, CustomMetadata custom, boolean indexing, null);
             }`
```

and method signature will be for the existing method with the addition of a new parameter.
```
public Document createDocument(OperationContext octxt, int folderId, ParsedDocument pd, MailItem.Type type,
            int flags, MailItem parent, CustomMetadata custom, boolean indexing, String nodeId)
```

Comment (shrutig0510) [reply store/src/java/com/zimbra/cs/mailbox/Mailbox.java:9502]: Done

**Review by vishnukantjangid — APPROVED** [review]

**Review by zimsuchitgupta — APPROVED** [review]

**Pull request merged**